security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -363,8 +368,13 @@ func (s *sdsservice) FetchSecrets(ctx context.Context, discReq *xdsapi.Discovery
 	if !s.skipToken {
 		t, err := getCredentialToken(ctx)
 		if err != nil {
-			sdsServiceLog.Errorf("Failed to get credential token: %v", err)
-			return nil, err
+			// Running in-process, no need to pass the token from envoy to agent as in-context - use the file
+			tok, err := ioutil.ReadFile("./var/run/secrets/tokens/istio-token")
```

Comment:
Same here. Falling back to local JWT should only happen when the agent is running in sidecar mode.